### PR TITLE
Prefer conda package over pip dependencies

### LINKED ISSUE #1129: [FEA]: Prefer conda packages and remove as many pip packages as possible from conda environment
### Is this a new feature, an improvement, or a change to existing functionality?

Change

### How would you describe the priority of this feature request

Medium

### Please provide a clear description of problem this feature solves

We have several pip packages being installed in our conda environment which have matching conda packages. Or they have dependencies which could be installed by conda. In almost every situation, we should prefer conda over pip in our conda environment config files.

The main driver for this is if a package is installed by pip and the same package but a different version is installed by conda, then the two packages collide and neither works correctly. Since pip does not do any dependency resolution, we should prefer conda (even with all of its problems).

### Describe your ideal solution

This issue should be resolved by performing the following:

- Search all packages being installed by pip and check if there is a matching conda package. If so, move it to the conda environment. 
- Any packages which cannot be installed by conda, check if there are any major dependencies which could be moved to the conda environment
   - This can be verified by performing a dry run of a pip install and see what packages it would have installed. Check to see if there are conda packages of those.

In the end, the number of packages that are installed by pip should be **_very low_**.

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by cwharris**
Prefer conda package over pip dependencies

Todo:
- [x] build and test inside devcontainer
- [x] build and test inside development container
- [ ] build and test inside release container

Closes #1129

### CODE REVIEW

File: ci/runner/Dockerfile [hunk 1]
```diff
@@ -67,6 +67,7 @@ COPY ./docker/conda/environments/cuda${CUDA_SHORT_VER}_examples.yml /tmp/conda/c
 COPY ./ci/scripts/download_kafka.py /tmp/scripts/download_kafka.py
 
 # Install extra deps needed for gnn_fraud_detection_pipeline & ransomware_detection examples
+# can stellargraph be obtained through conda?
```
Comment (mdemoret-nv) [inline ci/runner/Dockerfile:70]: Is this comment intentional?

File: docker/conda/environments/cuda11.8_dev.yml [hunk 2]
```diff
@@ -82,24 +86,18 @@ dependencies:
...
+    - tritonclient=2.26 # Required by NvTabular, force the version, so we get protobufs compatible with 4.21
     - typing_utils=0.1
     - watchdog=2.1
+    - websockets
     - yapf=0.40.1
-    ####### Morpheus Pip Dependencies (keep sorted!) #######
-    - pip:
-        # Add additional dev dependencies here
-        - docutils
-        - pytest-kafka==0.6.0
```
Comment (mdemoret-nv) [inline docker/conda/environments/cuda11.8_dev.yml:103]: What happened to `pytest-kafka`?

**Review by mdemoret-nv — COMMENTED** [review]
Couple of questions.

File: ci/runner/Dockerfile (hunk 1 above)
Comment (cwharris) [reply ci/runner/Dockerfile:70]: I saw we were using pip to install stellargraph here and left the comment so I wouldn't forget to check on it later. Guess it worked, in a round-about way. Should we try to install stellargraph via conda and get rid of this pip install call?

File: docker/conda/environments/cuda11.8_dev.yml (hunk 2 above)
Comment (cwharris) [reply docker/conda/environments/cuda11.8_dev.yml:103]: Good question. CI passes. Does it need to be installed in release? There's a script to install it in the devcontainer.

File: ci/runner/Dockerfile (hunk 1 above)
Comment (mdemoret-nv) [reply ci/runner/Dockerfile:70]: Lets leave it alone for now. There is a PR: #1032 which will remove stellargraph entirely. No need to worry about it for now.

File: docker/conda/environments/cuda11.8_dev.yml (hunk 2 above)
Comment (mdemoret-nv) [reply docker/conda/environments/cuda11.8_dev.yml:103]: `pytest-kafka` is needed for running the tests. We install it manually in the Dockerfile to ensure its setup for CI but this doesnt cover normal developer use. I think it should be left in the dev dependencies.

**Comment by cwharris** [comment]
@mdemoret-nv ready for re-review

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**